### PR TITLE
Add SettingsLayout to Groovy DSL doc

### PR DESCRIPTION
**Pull request by abstratt**
Also:

- tweak javadoc for SettingsLayout
- fix service scope for SettingsLayout

Issue: #16506

### CODE REVIEW

File: subprojects/core-api/src/main/java/org/gradle/api/file/SettingsLayout.java [hunk 1]
```diff
@@ -55,4 +58,27 @@ public interface SettingsLayout extends FileSystemLayout {
...
+     */
+    @Override
+    Provider<RegularFile> file(Provider<File> file);
+
+    /**
+     * {@inheritDoc}
+     *
+     * <p>
+     * File system locations based on relative paths will be
+     * resolved against this layout's reference location, as defined by {@link #getSettingsDirectory()}.
```
Comment (alllex) [inline subprojects/core-api/src/main/java/org/gradle/api/file/SettingsLayout.java:79]: 🎉  Thanks for doing this!

File: subprojects/core-api/src/main/java/org/gradle/api/file/ProjectLayout.java [hunk 2]
```diff
@@ -47,13 +47,24 @@ public interface ProjectLayout extends FileSystemLayout {
 
     /**
      * {@inheritDoc}
+     *
+     *  <p>
+     *  File system locations based on relative paths will be
+     *  resolved against this layout's reference location, as defined by {@link #getProjectDirectory()}.
+     *  </p>
```
Comment (alllex) [inline subprojects/core-api/src/main/java/org/gradle/api/file/ProjectLayout.java:54]: 💅  There seems to be an extra space in front of every javadoc line here

File: subprojects/core-api/src/main/java/org/gradle/api/file/ProjectLayout.java [hunk 3]
```diff
@@ -47,13 +47,24 @@ public interface ProjectLayout extends FileSystemLayout {
...
     Provider<RegularFile> file(Provider<File> file);
 
     /**
      * {@inheritDoc}
      *
+     * <p>
+     * File system locations based on relative paths will be
+     * resolved against this layout's reference location, as defined by {@link #getProjectDirectory()}.
+     * </p>
+     *
```
Comment (alllex) [inline subprojects/core-api/src/main/java/org/gradle/api/file/ProjectLayout.java:67]: ❔  Would the javadoc produce the same result if we drop the empty line 62 and the `</p>` line 66?

**Review by alllex — APPROVED** [review]
Happy to see the more detailed javadocs! 👍

File: subprojects/core-api/src/main/java/org/gradle/api/file/ProjectLayout.java (hunk 3 above)
Comment (abstratt) [reply subprojects/core-api/src/main/java/org/gradle/api/file/ProjectLayout.java:67]: Yup, closing paragraph tags are optional, and new lines have no effect on javdoc output. Are you suggesting we did that?

File: subprojects/core-api/src/main/java/org/gradle/api/file/ProjectLayout.java (hunk 2 above)
Comment (abstratt) [reply subprojects/core-api/src/main/java/org/gradle/api/file/ProjectLayout.java]: ✅

File: subprojects/core-api/src/main/java/org/gradle/api/file/ProjectLayout.java (hunk 3 above)
Comment (abstratt) [reply subprojects/core-api/src/main/java/org/gradle/api/file/ProjectLayout.java]: ✅

**Comment by abstratt** [comment]
@bot-gradle test and merge

File: subprojects/core-api/src/main/java/org/gradle/api/file/ProjectLayout.java (hunk 3 above)
Comment (alllex) [reply subprojects/core-api/src/main/java/org/gradle/api/file/ProjectLayout.java]: If it uses fewer lines of code, does the same thing, and does not hurt readability, I would take it any day! 😄

**Review by lkasso — APPROVED** [review]

**Pull request merged**